### DISCUSSION TITLE
Bug or feature: Async client components

### DISCUSSION THREAD

**Discussion by polarisdigitech** [vercel/next.js#50529] [App Router]
According to the feature set and documentation for the new app router, server components do data fetching like this:

```ts
export default async function Page({ searchParams }: PageProps) {
  const { page, size } = searchParams;

  const data = await getData(page, size);

  return <div></div>;
}
```

while client components do data fetching the usual way you do data fetching in React. But the example below works for client components:

```ts
"use client";

export default async function ModalRoute() {
  const searchParams = useSearchParams();

  const page = searchParams.get('page');
  const size = searchParams.get('size');

  const data = await getData(page, size);

  return <div></div>;
}
```

The component actual loads the data without wrapping it in a suspense boundary etc. But I noticed the component gets rendered twice. Is that a bug or a feature? I encountered the above example in the context of a modal/parallel route.

I assumed that modal/parallel routes are supposed to be client components, given there's a corresponding server component to render that same route during SSR.

And what best practices should we follow for modal/parallel routes?

**Comment by Uy575** [comment]
In docs they said if you are fetching data in server and then same way in the client side it will cause an issue you can install use server and write above same as use client then it will be good you might be facing issue for this.

**Reply by polarisdigitech** [reply]
But I didn't experience any issue, hence why I started this discussion

**Reply by Uy575** [reply]
they said it will not behave normally, you can read use server in docs may be it will help you.

**Reply by polarisdigitech** [reply]
I just told you I tried it and it works fine with no issue. Are you trolling or what?

**Reply by Uy575** [reply]
i understand but i am saying you can read that part may be you will get the ans if they provided any feature like this they discussed suspense  in very brief
